Engine/ac/draw.cpp

```diff
@@ -1087,6 +1087,34 @@ void draw_gui_sprite(Bitmap *ds, bool use_alpha, int x, int y, Bitmap *sprite, b
     }
 }
 
+void draw_gui_sprite_flipped(Bitmap *ds, int pic, int x, int y, bool use_alpha, BlendMode blend_mode, bool is_flipped)
+{
+    draw_gui_sprite_flipped(ds, use_alpha, x, y, spriteset[pic],
+        (game.SpriteInfos[pic].Flags & SPF_ALPHACHANNEL) != 0, blend_mode, 0xFF, is_flipped);
+}
+
+void draw_gui_sprite_flipped(Bitmap *ds, bool use_alpha, int x, int y, Bitmap *sprite, bool src_has_alpha,
+    BlendMode blend_mode, int alpha, bool is_flipped)
+{
+    if (alpha <= 0)
+        return;
+
+    if (is_flipped) {
+        Bitmap *tempspr = BitmapHelper::CreateTransparentBitmap(sprite->GetWidth(), sprite->GetHeight(), sprite->GetColorDepth());
```

Comment:
You do not delete tempspr below, causing a memory leak, I propose using `std::unique_ptr<Bitmap>` to make this automatic.

Reply:
I admit I'm always at loss on how to use these internal APIs

Reply:
> I admit I'm always at loss on how to use these internal APIs

Well, I must add more comments to these functions.

Reply:
I just found out this solution can cause garbage data. 
Maybe the unique_ptr assignment `sprite = tempspr.get();` , which triggers a delete, is causing this?